File: nms/src/main/java/com/hibiscusmc/hmcleaves/nms/NMSHandler.java
package com.hibiscusmc.hmcleaves.nms;

import org.bukkit.block.Block;
import org.bukkit.entity.Player;
import org.bukkit.inventory.ItemStack;

public interface NMSHandler {

    byte calculateBreakSpeed(Player player, ItemStack itemStack, Block block, int ticksPassed);

}

File: common/src/main/kotlin/com/hibiscusmc/hmcleaves/nms/NMSHandler.kt
package com.hibiscusmc.hmcleaves.nms

import com.hibiscusmc.hmcleaves.world.Position
import org.bukkit.entity.Player
import org.bukkit.inventory.EquipmentSlot

interface NMSHandler {

    fun handleRightClickCustomBlock(player: Player, placePosition: Position, slot: EquipmentSlot)

}